visitors/go_ast_pair/ast_pair_visitor.go

```diff
@@ -0,0 +1,294 @@
+package go_ast_pair
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"unicode"
+
+	. "github.com/akitasoftware/akita-libs/visitors"
+)
+
+// Structurally recurses through `leftNode` and `rightNode` in tandem.
```

Comment:
Is there a comment somewhere that describes what the pair visitor does when there are differences between the structures of the two nodes?

It looks like it descends until it finds a difference (either left or right are missing a node, or the nodes are of different types), and visits those nodes but not their children?